add PyAbel

DanHickstein opened the pull request:
One of the users of the PyAbel package has requested that we make the package available for conda install (https://github.com/PyAbel/PyAbel/issues/182). I believe that adding the package to conda-forge is the best way to do this.

I've created a ``meta.yml`` recipe based on the example file. Is this all that we need to do?

Thanks!

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/pyabel```) and found some lint.

Here's what I've got...


For **recipes/pyabel**:

 * Failed to even lint the recipe (might be a conda-smithy bug) :cry:

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/pyabel```) and found it was in an excellent condition.

jakirkham commented on recipes/pyabel/meta.yaml:
```diff
@@ -0,0 +1,60 @@
+    - numpy >=1.6
+    - setuptools >=16.0
+    - scipy >=0.14
+    - six >=1.10.0
```
Are all of these needed at build time or only at install time? Also is this linking against `cython`?

jakirkham replied:
Are all of these needed at build time or only at install time? Also is this linking against `numpy`?

rth replied:
@jakirkham  Thaks for the review! I would say `cython`, `toolchain` is only needed at build time,  `setuptools`, `cython` is needed to run the `setup.py`, the rest  is imported when running this package. We `cimport numpy` from [a Cython script](https://github.com/PyAbel/PyAbel/blob/a4ecf6a39f607e9e15cd0d968d2956c12f4a2393/abel/lib/direct.pyx) so I imagine this does link against numpy? We are also linking against `libm.so` (to use `libc.math` from Cython), I'm not sure if needs to be added as an explicit dependency or if it is built-in?

jakirkham replied:
For build time we mean whatever is required to run the build script (one line in this case). So it sounds like `cython`, `toolchain`, `setuptools`, and `numpy`. The rest of these can be cut from `build` then.

Also `cimport numpy` in Cython does mean linking to `numpy`. So we need `numpy x.x` in the `build` and `run` requirements.

`libm` is a system library. So there is nothing special for you to do there.

jakirkham commented on recipes/pyabel/meta.yaml:
```diff
@@ -0,0 +1,60 @@
+about:
+  home: https://github.com/PyAbel/PyAbel
+  license: MIT
+  license_family: MIT
```
Please add a PR to upstream to add `LICENSE` to `MANIFEST.in` so that it will be included in packages in the future. Also please add `license_file: LICENSE` with a comment. This can be uncommented in the future to package the license file.

jakirkham replied:
Done below.

jakirkham replied:
Note: Here is the PR ( https://github.com/PyAbel/PyAbel/pull/184 ) for reference.

jakirkham commented on recipes/pyabel/meta.yaml:
```diff
@@ -0,0 +1,60 @@
+    - DanHickstein
+    - rth
+    - stggh
+    - DhrubajyotiDas
```
@rth @stggh @DhrubajyotiDas , are you ok being included as maintainers of PyAbel on [conda-forge]( http://conda-forge.github.io/ )?

rth replied:
@jakirkham Fine with me.

stggh replied:
Fine with me too

DhrubajyotiDas replied:
Fine with me too!

jakirkham commented on recipes/pyabel/meta.yaml:
```diff
@@ -0,0 +1,60 @@
+{% set name = "PyAbel" %}
+{% set version = "0.7.4" %}
+{% set md5 = "1f9be2ecdc1ea8424000d62e2d6a4b10" %}
```
Please use sha256.

ref: https://pypi.org/

DanHickstein replied:
done

jakirkham commented on recipes/pyabel/meta.yaml:
```diff
@@ -0,0 +1,60 @@
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  md5: {{ md5 }}
```
Same as above.

DanHickstein commented:
Thanks for the help!!

I made the suggested changes. I'll also make a PR for PyAbel to include the license file in the MANIFEST.in.

DhrubajyotiDas commented on recipes/pyabel/meta.yaml:
```diff
@@ -57,4 +54,3 @@ extra:
     - DanHickstein
     - rth
     - stggh
-    - DhrubajyotiDas
```
Apologies for the late response, I'm "opting in".

scopatz commented on recipes/pyabel/meta.yaml:
```diff
@@ -0,0 +1,56 @@
+    - setuptools
+    - toolchain
+    - cython
+    - numpy >=1.6
```
Shouldn't this be `numpy x.x`?

mwcraig replied:
That depends on whether there are compiled extensions that link to numpy. If yes, the requirement should be listed twice:

```
- numpy x.x
- numpy >=1.6
```

in both build and run. Otherwise just the numpy version is fine.

DanHickstein replied:
What does `- numpy x.x` do?

@rth knows the Cython side of PyAbel the best, so he may have input. But yes, we have `cimport numpy as np`, so I think that means that we have compiled extensions that link to numpy.

rth replied:
@DanHickstein It's explained [here](https://github.com/conda-forge/staged-recipes#4-what-does-numpy-xx-mean). We are indeed linking against numpy, so we would have to add both lines I imagine..

DanHickstein replied:
Ha! Clearly I didn't read the FAQ very thoroughly :). I'll stick that line in both the build and the run.

jakirkham replied:
It basically just pins the version of NumPy that is installed with your package. That way, if you built this against NumPy 1.10, it wouldn't permit NumPy 1.11 being installed at the same time. This is a much better error than say a missing symbol error/crash that would occur otherwise if used with the wrong NumPy.

jakirkham replied:
IMHO this is the only remaining issue.

DanHickstein replied:
Thanks for your help! I just pushed a new version with the change made.

DanHickstein commented:
I made a new release of PyAbel (0.7.5), incorporating the change to MANIFEST.in that includes the license file. It should show up on PyPi in a few minutes, and then I can try to update this PR to reflect the new version. (So, don't merge quite yet...)

DanHickstein commented:
Okay, I got PyAbel 0.7.5 on PyPi, which now includes the license file. I updated the `meta.yml` to the new version, and uncommented `license_file: LICENSE.txt`.

So, as long as the builds go okay, we should be good-to-go!

DanHickstein commented:
Oh darn. AppVeyor uses the 2008 version of the microsoft compiler for Py3.4 as well. We need to disable building the Cython extensions for Py3.4 so that we don't get a stupid error about the `acosh` function.

DanHickstein commented:
We fixed the issues with the compilation of Cython extensions. (We removed the import of `acosh`, since not all compilers (MSVC2008 & 2010, I am looking at you) support this function.)

So, I think that we are good-to-go!

jakirkham commented:
Thanks everyone.
